core/commands/tags.py

```diff
@@ -0,0 +1,318 @@
+        regions.append(get_region(view_text))
+
+        self.remotes = list(self.get_remotes().keys())
+        if self.remotes:
+            for remote in self.remotes:
+                remote_text = REMOTE_TEMPLATE.format(remote, NO_REMOTE_TAGS_MESSAGE)
+                regions.append(get_region(remote_text))
+                view_text += remote_text
+
+        contents = header + view_text + KEY_BINDINGS_MENU
+
+        return contents, tuple(regions)
+
+
+class GsTagsFocusEventListener(EventListener):
+
+    """
+    If the current view is a tags view, refresh the view with
+    the repository's tags when the view regains focus.
+    """
+
+    def on_activated(self, view):
+        if view.settings().get("git_savvy.tags_view") == True:
+            view.run_command("gs_tags_refresh")
+
+
+class GsTagDeleteCommand(TextCommand, GitCommand):
+
+    """
+    Delete tag(s) in selection.
+    """
+
+    def run(self, edit):
+        # Valid sections are in the Local section
+        valid_ranges = view_section_ranges[self.view.id()][:1]
+
+        lines = util.view.get_lines_from_regions(
+            self.view,
+            self.view.sel(),
+            valid_ranges=valid_ranges
+            )
+
+        items = tuple(line[4:].strip().split() for line in lines if line)
+        if items:
+            for item in items:
+                self.git("tag", "-d", item[1])
+
+            util.view.refresh_gitsavvy(self.view)
+            sublime.status_message(TAG_DELETE_MESSAGE)
+
+
+class GsTagCreateCommand(WindowCommand, GitCommand):
+
+    """
+    Through a series of panels, allow the user to add a tag and message.
+    """
+
+    def run(self):
+        sublime.set_timeout_async(self.run_async)
+
+    def run_async(self):
+        """
+        Prompt the user for a tag name.
+        """
+        self.window.show_input_panel(
+            TAG_CREATE_PROMPT,
+            "",
+            self.on_entered_tag,
+            None,
+            None
+            )
+
+    def on_entered_tag(self, tag_name):
+        """
+        After the user has entered a tag name, prompt the user for a
+        tag message. If the message is empty, use the pre-defined one.
+        """
+        # If the user pressed `esc` or otherwise cancelled.
+        if not tag_name:
+            return
+
+        # TODO: do some validation
+
+        self.tag_name = tag_name
+        self.window.show_input_panel(
+            TAG_CREATE_MESSAGE_PROMPT,
+            "",
+            self.on_entered_message,
+            None,
+            None
+            )
+
+    def on_entered_message(self, message):
+        """
+        Perform `git tag tag_name -F -`
+        """
+        # If the user pressed `esc` or otherwise cancelled
+        if message == -1:
+            return
+
+        if not message:
+            default_message = sublime.load_settings("GitSavvy.sublime-settings").get("default_tag_message")
+            message = default_message.format(tag_name=self.tag_name)
```

Comment:
That would be more explicit.  I suppose it would be strange to be prompted for a message, you enter an empty string (for some reason) and it turns out that the message is not what you entered.  Providing a pre-filled default is a good idea, and probably better user experience.